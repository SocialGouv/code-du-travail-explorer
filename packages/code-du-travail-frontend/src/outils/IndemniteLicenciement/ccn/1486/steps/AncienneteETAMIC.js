import { Input, theme, Toast } from "@socialgouv/cdtn-react-ui";
import React from "react";
import { Field } from "react-final-form";
import styled from "styled-components";

import { CurrencyField } from "../../../../common/CurrencyField";
import { Error } from "../../../../common/ErrorField";
import { Label, SectionTitle } from "../../../../common/stepStyles";
import { isNumber } from "../../../../common/validators";
import { YesNoQuestion } from "../../../../common/YesNoQuestion";

export const CONTRAT_KEY = "brancheContrat";
export const HAS_CONTRAT_KEY = "hasBrancheContrat";

export const AncienneteETAMIC = ({ form }) => {
  const data = form.getState().values;
  const hasContrats = data[HAS_CONTRAT_KEY];

  return (
    <>
      <SectionTitle>Contrats antérieurs</SectionTitle>
      <p>
        Si le salarié a été auparavant employé par la même société et que la fin
        du / des contrat(s) n’est pas imputable à l’une des raisons
        suivantes&nbsp;:
      </p>
      <ul>
        <li>démission</li>
        <li>licenciement pour faute grave ou faute lourde</li>
      </ul>
      <p>
        alors{" "}
        <strong>
          la durée de ce(s) contrat(s) s’ajoute à votre ancienneté
        </strong>
        .
      </p>
      <StyledToast>
        Dans le cas de la démission, l’employeur peut accepter de prendre en
        compte la durée du contrat de travail correspondant pour l’appréciation
        de l’ancienneté.
      </StyledToast>
      <YesNoQuestion
        name={HAS_CONTRAT_KEY}
        label="Avez-vous un/des contrat(s) antérieur(s) à déclarer ?"
        onChange={(HAS_CONTRAT_KEY) => {
          HAS_CONTRAT_KEY
            ? form.change(CONTRAT_KEY, {
                considered: false,
                duration: "",
                indemnite: "0",
              })
            : form.change(CONTRAT_KEY, undefined);
        }}
      />
      {hasContrats && (
        <>
          <Field
            name={`${CONTRAT_KEY}.duration`}
            validate={isNumber}
            subscription={{
              error: true,
              invalid: true,
              touched: true,
              value: true,
            }}
            render={({ input, meta: { touched, error, invalid } }) => (
              <>
                <Label htmlFor="duration">
                  Durée des contrats de travail précédents (en mois)
                </Label>
                <Input
                  {...input}
                  id="duration"
                  type="number"
                  invalid={touched && invalid}
                />
                {error && touched && <Error>{error}</Error>}
              </>
            )}
          />
          <StyledToast>
            Il est important de préciser l’indemnité de licenciement
            éventuellement perçue lors d’un précédent licenciement dans la même
            entreprise. L’employeur actuel peut la déduire de l’indemnité de
            licenciement en train d’être calculée.
          </StyledToast>
          <CurrencyField
            name={`${CONTRAT_KEY}.indemnite`}
            label="Indemnité de licenciement précédemment perçue"
            required={false}
          />
          <YesNoQuestion
            name={`${CONTRAT_KEY}.considered`}
            label="Souhaitez vous déduire cette indemnité&nbsp;?"
            required={false}
          />
        </>
      )}
    </>
  );
};

const { spacings } = theme;

const StyledToast = styled(Toast)`
  margin-top: ${spacings.medium};
`;
