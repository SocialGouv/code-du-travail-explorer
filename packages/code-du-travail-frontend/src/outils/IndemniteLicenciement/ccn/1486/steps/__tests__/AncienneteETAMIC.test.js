import React from "react";
import { render } from "@testing-library/react";
import { Form } from "react-final-form";
import { AncienneteETAMIC } from "../AncienneteETAMIC.js";

describe("<AncienneteETAMIC />", () => {
  it("renders", () => {
    const onSubmit = jest.fn();
    const { container } = render(
      <Form onSubmit={onSubmit}>
        {({ form }) => <AncienneteETAMIC form={form} />}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
  it("asks to fill duration, indemnitÃ© and consideration", () => {
    const onSubmit = jest.fn();
    const initialValues = {
      hasBrancheContrat: true,
      brancheContrat: {
        duration: "3",
        indemnite: "0",
        considered: false,
      },
    };
    const { container } = render(
      <Form onSubmit={onSubmit} initialValues={initialValues}>
        {({ form }) => <AncienneteETAMIC form={form} />}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });
});
