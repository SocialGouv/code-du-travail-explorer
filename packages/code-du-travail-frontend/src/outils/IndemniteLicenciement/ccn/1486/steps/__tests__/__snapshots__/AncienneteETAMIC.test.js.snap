// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AncienneteETAMIC /> asks to fill duration, indemnité and consideration 1`] = `
.c3 {
  position: absolute;
  background-color: #7994d4;
  border-radius: 0.6rem;
  top: 50%;
  left: 0;
  width: 0.4rem;
  height: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c2 {
  position: relative;
  margin: 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
  padding-left: 3.2rem;
  text-align: left;
}

.c0 {
  margin: 0 auto 1rem;
  text-align: left;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: auto;
  min-height: 48px;
  color: #3e486e;
  background-color: #fff;
  border-color: #7994d4;
  border-style: solid;
  border-width: 2px;
  border-radius: 0.6rem;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 1rem 1.6rem;
  color: #3e486e;
  text-align: left;
}

.c11 {
  width: 10em;
  text-align: right;
  border-color: #bbcadf;
}

.c11::-webkit-outer-spin-button,
.c11::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c1 {
  margin-top: 3.2rem;
  margin-bottom: 2rem;
}

.c7 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c12 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: pointer;
}

.c8 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
}

.c14 {
  width: 10em;
  text-align: right;
  border-color: #bbcadf;
  padding-right: 1.6rem;
  text-align: right;
}

.c14::-webkit-outer-spin-button,
.c14::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c13 {
  position: relative;
  display: inline-block;
  margin-right: 1.6rem;
}

.c15 {
  position: absolute;
  top: 50%;
  right: 0.25rem;
  color: #4d73b8;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c5 {
  margin-top: 2rem;
}

@media (max-width:600px) {
  .c2 {
    padding-left: 1.6rem;
  }
}

@media (max-width:600px) {
  .c2 {
    font-size: 2.2rem;
  }
}

@media (max-width:600px) {
  .c7 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {
  .c12 {
    font-size: 1.6rem;
  }
}

<div>
  <header
    class="c0 c1"
  >
    <h2
      class="c2"
    >
      <div
        class="c3"
      />
      Contrats antérieurs
    </h2>
  </header>
  <p>
    Si le salarié a été auparavant employé par la même société et que la fin du / des contrat(s) n’est pas imputable à l’une des raisons suivantes :
  </p>
  <ul>
    <li>
      démission
    </li>
    <li>
      licenciement pour faute grave ou faute lourde
    </li>
  </ul>
  <p>
    alors
     
    <strong>
      la durée de ce(s) contrat(s) s’ajoute à votre ancienneté
    </strong>
    .
  </p>
  <div
    class="c4 c5"
  >
    <div
      class="c6"
      role="alert"
    >
      Dans le cas de la démission, l’employeur peut accepter de prendre en compte la durée du contrat de travail correspondant pour l’appréciation de l’ancienneté.
    </div>
  </div>
  <p
    class="c7"
  >
    Avez-vous un/des contrat(s) antérieur(s) à déclarer ?
    <span
      aria-label="champs obligatoire"
      class="c8"
    >
      *
    </span>
  </p>
  <div
    class="c9"
  >
    <label
      class="c10"
    >
      <input
        checked=""
        name="hasBrancheContrat"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c10"
    >
      <input
        name="hasBrancheContrat"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <label
    class="c10"
    for="duration"
  >
    Durée des contrats de travail précédents (en mois)
  </label>
  <input
    class="c11"
    id="duration"
    name="brancheContrat.duration"
    type="number"
    value="3"
  />
  <div
    class="c4 c5"
  >
    <div
      class="c6"
      role="alert"
    >
      Il est important de préciser l’indemnité de licenciement éventuellement perçue lors d’un précédent licenciement dans la même entreprise. L’employeur actuel peut la déduire de l’indemnité de licenciement en train d’être calculée.
    </div>
  </div>
  <label
    class="c12"
    for="currency-1"
  >
    Indemnité de licenciement précédemment perçue
  </label>
  <div
    class="c13"
  >
    <input
      class="c14"
      id="currency-1"
      name="brancheContrat.indemnite"
      type="number"
      value="0"
    />
    <span
      aria-hidden="true"
      class="c15"
    >
      €
    </span>
  </div>
  <p
    class="c7"
  >
    Souhaitez vous déduire cette indemnité ?
  </p>
  <div
    class="c9"
  >
    <label
      class="c10"
    >
      <input
        name="brancheContrat.considered"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c10"
    >
      <input
        checked=""
        name="brancheContrat.considered"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
</div>
`;

exports[`<AncienneteETAMIC /> renders 1`] = `
.c3 {
  position: absolute;
  background-color: #7994d4;
  border-radius: 0.6rem;
  top: 50%;
  left: 0;
  width: 0.4rem;
  height: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c2 {
  position: relative;
  margin: 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
  padding-left: 3.2rem;
  text-align: left;
}

.c0 {
  margin: 0 auto 1rem;
  text-align: left;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: auto;
  min-height: 48px;
  color: #3e486e;
  background-color: #fff;
  border-color: #7994d4;
  border-style: solid;
  border-width: 2px;
  border-radius: 0.6rem;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 1rem 1.6rem;
  color: #3e486e;
  text-align: left;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c1 {
  margin-top: 3.2rem;
  margin-bottom: 2rem;
}

.c7 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c8 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
}

.c5 {
  margin-top: 2rem;
}

@media (max-width:600px) {
  .c2 {
    padding-left: 1.6rem;
  }
}

@media (max-width:600px) {
  .c2 {
    font-size: 2.2rem;
  }
}

@media (max-width:600px) {
  .c7 {
    font-size: 1.6rem;
  }
}

<div>
  <header
    class="c0 c1"
  >
    <h2
      class="c2"
    >
      <div
        class="c3"
      />
      Contrats antérieurs
    </h2>
  </header>
  <p>
    Si le salarié a été auparavant employé par la même société et que la fin du / des contrat(s) n’est pas imputable à l’une des raisons suivantes :
  </p>
  <ul>
    <li>
      démission
    </li>
    <li>
      licenciement pour faute grave ou faute lourde
    </li>
  </ul>
  <p>
    alors
     
    <strong>
      la durée de ce(s) contrat(s) s’ajoute à votre ancienneté
    </strong>
    .
  </p>
  <div
    class="c4 c5"
  >
    <div
      class="c6"
      role="alert"
    >
      Dans le cas de la démission, l’employeur peut accepter de prendre en compte la durée du contrat de travail correspondant pour l’appréciation de l’ancienneté.
    </div>
  </div>
  <p
    class="c7"
  >
    Avez-vous un/des contrat(s) antérieur(s) à déclarer ?
    <span
      aria-label="champs obligatoire"
      class="c8"
    >
      *
    </span>
  </p>
  <div
    class="c9"
  >
    <label
      class="c10"
    >
      <input
        name="hasBrancheContrat"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c10"
    >
      <input
        name="hasBrancheContrat"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
</div>
`;
